src/binaryen-c.cpp

```diff
@@ -3511,6 +3511,37 @@ uint32_t BinaryenGetMemorySegmentByteOffset(BinaryenModuleRef module,
   Fatal() << "non-constant offsets aren't supported yet";
   return 0;
 }
+int BinaryenHasMemory(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.exists;
+}
+size_t BinaryenMemoryGetInitial(BinaryenModuleRef module) {
```

Comment:
Oh a further note -- this could have issues I think on 64-bit memories, as these values are Address instances and may be 64-bit on a 64-bit memory.

Reply:
We may at some point need to make this 64-bit, but for now this can use `uint32_t` which is consistent with other places an Address is used (like a Load offset). Eventually for wasm64 we'll need to do a pass on all those.